Align dependencies with lnp-node e81e693d1a92fc5aef10c423648db43e4b755e97

sedited opened the pull request:
The deps are alligned, but parallel swap test does not run yet. 

As part of this PR and besides the pure API changes arising from the deps bump I also changed:

- Peerd raw unencrypted connection to brontide encrypted connection
- `esb::SenderList`to `Endpoints`
- Renamed `senders` to the more appropriate `endpoints`
- Use `esb::Error<ServiceId>` instead of `esb::Error`
- Use lnp `Ping`, `Error` and `Init` messages instead of our own. This permanently binds us to the lnp::p2p crate

sedited committed:
Allign dependencies with lnp-node a403e2561ff3f24056ed969f75c4f06f958785f7

Patches for dependency updates:

Update clap usage

Update LocalNode usage

sedited committed:
Fix esb Error type

sedited committed:
Resolve new endpoint type

sedited committed:
Resolve moved lnp-core deps

sedited committed:
Fix error handling after dep allignment

sedited committed:
Fix clap import after dep allignment

TheCharlatan commented:
Needs merge of https://github.com/farcaster-project/farcaster-core/pull/215

Lederstrumpf committed:
bump rust version in CI to 1.59.0

Lederstrumpf committed:
bump farcaster_core to 0.4.4

Lederstrumpf reviewed (approved):
tACK

TheCharlatan commented:
Requires upstream patching in either internet2 or microservices. Tracking in https://github.com/LNP-WG/lnp-node/issues/50

sedited committed:
Peerd: patch build after rebase

sedited committed:
Deps: Wrap up dependency bump

sedited committed:
Peerd: Use brontide encryption for tcp session

Lederstrumpf committed:
remove dead code linking from CI

originally specified by codegen tool, but induces a recursion limit overflow in
bitcoin_scripts 0.7.0, so removing ("temporarily")

sedited committed:
Rebase fixup s/senders/endpoints/

TheCharlatan commented:
Seems like the CI decided to pass now, so I'll open it for review.

h4sh3d commented:
Let me push 1 change so the monero functional tests pass with the newest `monero-lws:latest`. They currently don't because the container crash at startup because the name of a var changed.
After review and merge I can better test https://github.com/farcaster-project/farcaster-node/pull/444 and see if there is no real regression.

h4sh3d committed:
fix: update monero lws env variable

h4sh3d committed:
fix: ran cargo msrv to find correct version

Lederstrumpf commented on Cargo.toml:
```diff
@@ -39,7 +39,9 @@ required-features = ["server"]
 # Farcaster crates
-farcaster_core = { version = "0.4.3", features = ["serde"] }
+# farcaster_core = { version = "0.4.4", features = ["serde"] }
+farcaster_core = { git = "https://github.com/farcaster-project/farcaster-core", branch = "main", features = ["serde"] }
```
I just did a release of core so that we don't have this wart anymore, but it failed since we currently have our own fork of secp256kfun in there. But I'm good to merge without this fixed.

Lederstrumpf reviewed (approved):
tACK @ `0f38acb` - thanks for having all your teeth pulled @TheCharlatan! ;)

TheCharlatan commented:
I can has merge?
